sunpy/timeseries/metadata.py

```diff
@@ -327,10 +348,47 @@ def columns(self):
         for metatuple in self.metadata:
             all_cols.update(metatuple[1])
         return list(all_cols)
-        
-    def remove_columns(self, colnames):
+
+    @property
+    def metas(self):
+        """Returns a list of all the metadict objects in the TimeSeriesMetaData object."""
+        all_metas = []
+        for metatuple in self.metadata:
+            all_metas.append(metatuple[2])
+        return all_metas
+
+    @property
+    def timeranges(self):
+        """Returns a list of all the TimeRange objects the TimeSeriesMetaData object."""
+        all_tr = []
+        for metatuple in self.metadata:
+            all_tr.append(metatuple[0])
+        return all_tr
+
+    def values(self):
+        """Returns a list of all the values from the metadict objects in each
+        entry in the TimeSeriesMetaData object."""
+        all_vals = set()
+        for metatuple in self.metadata:
+            for key, value in metatuple[2].items():
+                all_vals.add(str(value))
+        all_vals = list(all_vals)
+        all_vals.sort()
+        return all_vals
+
+    @property
+    def timerange(self):
+        """Returns the TimeRange of the entire time series meta data."""
+        start = self.metadata[0][0].start

```

Comment:
or you make .metadata a setter/property which sorts the data into a `._metadata` hidden variable.

like this:

```python
    @property
    def temperature(self):
        print("Getting value")
        return self._temperature

    @temperature.setter
    def temperature(self, value):
        if value < -273:
            raise ValueError("Temperature below -273 is not possible")
        print("Setting value")
        self._temperature = value
```

from http://www.programiz.com/python-programming/property